### PR TITLE
Use an organization variable for the Python testing matrix entries

### PR DESCRIPTION
**Pull request by webbnh**
## Changes introduced with this PR

This change replaces the hard-coded list of Python versions in the test `matrix` with a reference to the Arcalot organization variable, `ARCALOT_PYTHON_SUPPORTED_VERSIONS`.

This PR is waiting for the fix in https://github.com/arcalot/arcaflow-plugin-sdk-python/pull/135 to be released.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -11,7 +11,9 @@ jobs:
     name: Build
     strategy:
       matrix:
-        python-version: [ '3.9', '3.10', 'pypy3.9' ]
+        python-version: ${{ fromJSON(vars.ARCALOT_PYTHON_SUPPORTED_VERSIONS) }}
+        exclude:
+          - python-version: 3.12
```
Comment (webbnh) [inline .github/workflows/build.yaml:16]: We probably don't want this PR merged with this code.  Instead, we should either [fix the Python SDK](https://github.com/arcalot/arcaflow-lib-kubernetes-python/actions/runs/9654359931/job/26628436184#step:8:153), or remove `3.12` from `ARCALOT_PYTHON_SUPPORTED_VERSIONS`.

Thoughts, @jaredoconnell, @mfleader, @dbutenhof?

**Review by webbnh — COMMENTED** [review]
It looks like [the Python SDK is not ready for Python 3.12](https://github.com/arcalot/arcaflow-lib-kubernetes-python/actions/runs/9654359931/job/26628436184#step:8:153).

File: .github/workflows/build.yaml (hunk 1 above)
Comment (webbnh) [reply .github/workflows/build.yaml:16]: This was fixed in https://github.com/arcalot/arcaflow-plugin-sdk-python/pull/135; once that is released and the dependency here is updated, this PR can proceed without the above code.

**Review by mfleader — APPROVED** [review]

**Comment by mfleader** [comment]
I agree it seems good to go once the new release is created for the python sdk.

**Review by jaredoconnell — APPROVED** [review]
Good once the issue to remove the exclusion is completed following the upcoming release of the Python SDK.

**Review by jaredoconnell — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**